database/src/main/runs/get_segmentation_measurements.sql

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0 *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION runs.get_segmentation_measurements(
+    IN  i_segmentation                               HSTORE,
+    OUT status                                       INTEGER,
+    OUT status_text                                  TEXT,
+    OUT id_measurement                               UUID,
+    OUT id_checkpoint_measure_definition             BIGINT,
+    OUT id_checkpoint                                UUID,
+    OUT value                                        JSON,
+    OUT checkpoint_name                              TEXT,
+    OUT workflow_name                                TEXT,
+    OUT checkpoint_process_start_time                TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_process_end_time                  TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_created_by                        TEXT,
+    OUT checkpoint_created_at                        TIMESTAMP WITH TIME ZONE,
+    OUT measure_type                                 TEXT,
+    OUT measure_fields                               TEXT[],
+    OUT checkpoint_measure_definition_created_by     TEXT,
+    OUT checkpoint_measure_definition_created_at     TIMESTAMP WITH TIME ZONE
```

Comment:
Two remarks: what is returned and which order.
On the first one, I would return only values expected to be useful to the caller based on the projected use of the function. Particularly in the first iteration. It's easy to add fields later if needed.
Second, I would order the fields in some logical manner, from top to bottom, summary to detail.
So I would keep the fields:
`id_measurement, checkpoint_name, workflow_name, checkpoint_process_start_time, checkpoint_process_end_time,  measure_type, measure_fields, value`

Reply:
Updated.